device/src/state_machines/session.rs

```diff
@@ -485,6 +483,12 @@ where
                                                 && (fcnt > session.fcnt_down || fcnt == 0)
                                             {
                                                 session.fcnt_down = fcnt;
+                                                // increment the FcntUp since we have received
+                                                // downlink - only reason to not increment
+                                                // is if confirmed frame is sent and no
+                                                // confirmation (ie: downlink) occurs
+                                                session.fcnt_up_increment();
```

Comment:
am I right to think that here is an assumption that the device is a class A device and if so is it an intended assumption and shouldn't it be documented?

Reply:
Yes, that's absolutely an assumption! I'll add it to the docs - good point